src/pages/iou/MoneyRequestTagPage.js

```diff
@@ -96,26 +95,14 @@ MoneyRequestTagPage.displayName = 'MoneyRequestTagPage';
 MoneyRequestTagPage.propTypes = propTypes;
 MoneyRequestTagPage.defaultProps = defaultProps;
 
-export default compose(
-    withOnyx({
-        iou: {
-            key: ONYXKEYS.IOU,
-        },
-    }),
-    // eslint-disable-next-line rulesdir/no-multiple-onyx-in-file
-    withOnyx({
-        report: {
-            key: ({route, iou}) => {
-                const reportID = IOU.getIOUReportID(iou, route);
-
-                return `${ONYXKEYS.COLLECTION.REPORT}${reportID}`;
-            },
-        },
-    }),
-    // eslint-disable-next-line rulesdir/no-multiple-onyx-in-file
-    withOnyx({
-        policyTags: {
-            key: ({report}) => `${ONYXKEYS.COLLECTION.POLICY_TAGS}${report ? report.policyID : '0'}`,
-        },
-    }),
-)(MoneyRequestTagPage);
+export default withOnyx({
+    report: {
+        key: ({route, iou}) => `${ONYXKEYS.COLLECTION.REPORT}${IOU.getIOUReportID(iou, route)}`,
+    },
+    iou: {
+        key: ONYXKEYS.IOU,
+    },
```

Comment:
I noticed a slight difference.

Multiple withOnyx:

<img width="1141" alt="Screenshot 2023-11-09 at 10 18 59 PM" src="https://github.com/Expensify/App/assets/108292595/dff160f9-6362-456e-8f0b-25dd42403769">

Single withOnyx:

<img width="1143" alt="Screenshot 2023-11-09 at 10 17 07 PM" src="https://github.com/Expensify/App/assets/108292595/20459a37-4599-48ce-8e9b-cd1b0bb4f838">

As you see, `iou` is undefined first time

Reply:
Ah, that's interesting. The `undefined` is expected so the only thing that matters is that `getIOUReport()` protects against an `undefined` value, which it does. However, I see that it returns an empty string in the case of `undefined` which is not desired and would subscribe to the entire report collection. I've updated this to return `0` for the iou report ID which is the more proper way of handling this.

Reply:
I am still concerned about `undefined` on first render.
Sometimes this will cause unexpected behavior like https://github.com/Expensify/App/pull/30866#issuecomment-1804517297

Reply:
Yeah, I agree with you that it's a problem for https://github.com/Expensify/App/pull/30866. I would like to take a look at it in that PR and try to solve it over there. I feel it's unrelated to this specifically, but will let you know.